web/app/js/components/Deployments.jsx

```diff
@@ -60,35 +67,93 @@ export default class Deployments extends React.Component {
     this.setState({ pendingRequests: true });
 
     let rollupPath = `${this.props.pathPrefix}/api/metrics?window=${this.state.metricsWindow}`;
-    let timeseriesPath = `${rollupPath}&timeseries=true`;
     let podPath = `${this.props.pathPrefix}/api/pods`;
     let rollupRequest = fetch(rollupPath).then(r => r.json());
-    let timeseriesRequest = fetch(timeseriesPath).then(r => r.json());
+
     let podRequest = fetch(podPath).then(r => r.json());
 
-    Promise.all([rollupRequest, timeseriesRequest, podRequest])
-      .then(([metrics, ts, p]) => {
+    Promise.all([rollupRequest, podRequest])
+      .then(([rollup, p]) => {
+        let poByDeploy = this.getDeploymentList(p.pods);
+        let meshDeploys = _.compact(processRollupMetrics(rollup.metrics, "targetDeploy"));
+        let combinedMetrics = this.addDeploysWithNoMetrics(poByDeploy, meshDeploys);
+
+        let limitSparklineData = _.size(meshDeploys) > maxTsToFetch;
+        this.loadTimeseriesFromServer(meshDeploys);
```

Comment:
I'm not positive, but am seeing some state issues with this component, and I think it stems from the fact that `loadTimeseriesFromServer` sets state, and then this method also sets state immediately after this call. I think it would better if  `loadTimeseriesFromServer` returned an object, and then that object was used to modify state in this method, such that the state is only changing once per call to `loadFromServer`.

Reply:
yeah originally I didn't want to block rendering the rest of the page on loading timeseries (so that some data would render on the page while we're waiting for a slower call) but now that we're limiting the ts requests anyway, we could.